src/client/Microsoft.Identity.Client/Internal/Broker/BrokerInteractiveRequestComponent.cs

```diff
@@ -83,7 +83,12 @@ public async Task<MsalTokenResponse> FetchTokensAsync(CancellationToken cancella
             {
                 _logger.Info(
                     LogMessages.ErrorReturnedInBrokerResponse(msalTokenResponse.Error));
-                throw new MsalServiceException(msalTokenResponse.Error, MsalErrorMessage.BrokerResponseError + msalTokenResponse.ErrorDescription);
+
+                throw MsalServiceExceptionFactory.FromBrokerResponse(msalTokenResponse.Error,
```

Comment:
Do we always get Error, SubError etc. ? If not, we should add some sensible defaults, smth like "unkown_broker_error" "Something happened. Please collect broker logs". 

Currently if we don't know the error details we throw smth related to AndroidBrokerCannotBeInvoked, which is very misleading.